### ISSUE TITLE
Do not use network from main thread

### ISSUE THREAD

**Issue by nitehu** [ogarcia/ultrasonic#631]
## Problem description

I've seen the exception below in the logs.
I think somewhere there is a network operation on the main thread which fails.

```
D/NavigationActivity: Navigated to org.moire.ultrasonic.debug:id/playerFragment
D/PlayerFragment: EqualizerController Observer.onChanged received controller
D/TimberOkHttpLogger: --> GET http://192.168.0.101:4040/rest/getUser.view?username=admin&u=admin&c=Ultrasonic&f=json&v=1.15.0&t=ff4d54dfbbc2cc8bdaa516e74fb02240&s=63B89DF2131AA0E406C6CCACCD81923A
    --> END GET
D/TimberOkHttpLogger: <-- HTTP FAILED: android.os.NetworkOnMainThreadException
W/MediaPlayerController: Error getting user information
    android.os.NetworkOnMainThreadException
        at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1605)
        at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:116)
        at java.net.SocketOutputStream.write(SocketOutputStream.java:161)
        at okio.Okio$1.write(Okio.java:79)
        at okio.AsyncTimeout$1.write(AsyncTimeout.java:180)
        at okio.RealBufferedSink.flush(RealBufferedSink.java:224)
        at okhttp3.internal.http1.Http1Codec.finishRequest(Http1Codec.java:166)
        at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:84)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
W/MediaPlayerController:     at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:225)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at org.moire.ultrasonic.api.subsonic.interceptors.RangeHeaderInterceptor.intercept(RangeHeaderInterceptor.kt:33)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at org.moire.ultrasonic.api.subsonic.interceptors.PasswordMD5Interceptor.intercept(PasswordMD5Interceptor.kt:29)
        at org.moire.ultrasonic.api.subsonic.interceptors.ProxyPasswordInterceptor.intercept(ProxyPasswordInterceptor.kt:26)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at org.moire.ultrasonic.api.subsonic.interceptors.VersionInterceptor.intercept(VersionInterceptor.kt:28)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at org.moire.ultrasonic.api.subsonic.SubsonicAPIClient.okHttpClient$lambda-1(SubsonicAPIClient.kt:76)
        at org.moire.ultrasonic.api.subsonic.SubsonicAPIClient.lambda$Ur6c_X6tMhnf998uNKb9-VLSBEs(Unknown Source:0)
        at org.moire.ultrasonic.api.subsonic.-$$Lambda$SubsonicAPIClient$Ur6c_X6tMhnf998uNKb9-VLSBEs.intercept(Unknown Source:2)
W/MediaPlayerController:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
        at okhttp3.RealCall.execute(RealCall.java:93)
        at retrofit2.OkHttpCall.execute(OkHttpCall.java:188)
        at retrofit2.DefaultCallAdapterFactory$ExecutorCallbackCall.execute(DefaultCallAdapterFactory.java:104)
        at org.moire.ultrasonic.service.RESTMusicService.getUser(RESTMusicService.kt:585)
        at org.moire.ultrasonic.service.CachedMusicService.getUser(CachedMusicService.kt:437)
        at org.moire.ultrasonic.service.MediaPlayerController.isJukeboxAvailable(MediaPlayerController.kt:420)
        at org.moire.ultrasonic.fragment.PlayerFragment$onViewCreated$23.invokeSuspend(PlayerFragment.kt:434)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:367)
        at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:30)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.startCoroutineImpl(Builders.common.kt:192)
        at kotlinx.coroutines.BuildersKt.startCoroutineImpl(Unknown Source:1)
        at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:134)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
        at kotlinx.coroutines.BuildersKt.launch(Unknown Source:1)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
        at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source:1)
        at org.moire.ultrasonic.fragment.PlayerFragment.onViewCreated(PlayerFragment.kt:432)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:2974)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:543)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2177)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2094)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1990)
        at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Comment by tzugen** [comment]
It's a  bug in the PR #628 
I'll push a fix in a bit

**Comment by nitehu** [comment]
Oh okay, I thought it was an older bug :)
Edit: no really, it is older, I've found it during testing #626

**Comment by tzugen** [comment]
Ok, yes, you are right :)
I touched the same place in the PR and thought it came from the PR.
I added the fix to #628

**Comment by tzugen** [comment]
Ok, yes, you are right :)
I touched the same place in the PR and thought it came from the PR.
I added the fix to #628
